cli/server/TileCache.js

```diff
@@ -0,0 +1,133 @@
+const _ = require('lodash');
+const { default: fetch, Headers } = require('node-fetch');
+const fs = require('fs-extra');
+const path = require('path');
+
+const utils = require('../utils');
+
+const TILES_CACHE_DIR = 'cache/tiles';
+
+
+function areValidCoordinates(coord) {
+  return _.every(coord, (c) => _.isInteger(_.toNumber(c)));
+}
+
+/**
+ * Caches map tiles requested by the client, so that previously seen tiles are
+ * available from localhost.
+ */
+class TileCache {
+  constructor() {
+    // TODO: setup transpiling server code to avoid .bind(this) and for other syntactic perks
```

Comment:
I'm not a fan of transpiling the server code as I value the simplicity of being able to run `node server.js`, although it does mean some code is a little bit more verbose. For background, auspice used to transpile the server code and it's been a much nicer dev experience since we stopped doing this.